ext/opcache/jit/zend_jit_x86.dasc

```diff
@@ -8966,18 +8967,12 @@ static int zend_jit_recv(dasm_State **Dst, const zend_op *opline, zend_op_array
 					|	add r0, offsetof(zend_reference, val)
 				}
 				if (!ZEND_TYPE_IS_CLASS(type)) {
-					unsigned char code = ZEND_TYPE_CODE(type);
-
-					if (code == _IS_BOOL) {
-						|	cmp byte [r0 + 8], IS_FALSE
-						|	je >1
-						|	cmp byte [r0 + 8], IS_TRUE
-						|	jne >8
-						|1:
-					} else {
-						|	cmp byte [r0 + 8], code
-						|	jne >8
-					}
+					uint32_t type_mask = ZEND_TYPE_MASK(type);
+					|	mov edx, 1
+					|	mov cl, byte [r0 + 8]
+					|	shl edx, cl
+					|	test edx, type_mask
+					|	je >8
```

Comment:
@dstogov I'm always unsure what registers I'm allowed to use ... is the use of `cl` and `edx` here okay?

Reply:
It's OK to use scratch registers. See https://wiki.osdev.org/Calling_Conventions
eax, ecx, edx - are OK.